R/extract.R

```diff
@@ -55,7 +55,9 @@ str_extract <- function(x, into, regex, convert = FALSE) {
     is_character(into)
   )
 
-  matches <- stringi::stri_match_first_regex(x, regex)[, -1, drop = FALSE]
+  matches <- lapply(regmatches(x, regexec(regex, x)),
```

Comment:
Could you please pull this out into a separate function? It's going to need unit tests for edge cases that `stri_match_first_regex()` might handle, but the base methods do not. I don't know exactly what those are, but I'd start with tests around no matches, NA matches, ...

Reply:
Pulled it out and added a test suite. Based it off of stringi and adjusted to match. stringi had a lot of tests that worked around capture groups and blanks however tidyr won't need it as it checks earlier in extract to ensure the number of capture groups is consistent with the into value